crawler/urlJobProcessor.js

```diff
@@ -1,36 +1,106 @@
 import { default as fetch } from 'node-fetch';
+import { default as cheerio } from 'cheerio';
+import { default as urlModule } from 'url';
 
 const probes = [];
 
-function configure(/* opts */) {
-  // TODO: Set up probes. They'll be specified somehow in opts and we
-  // should be able to load each probe individually. Maybe something
-  // like `require('./probe-' + probeName + '.js');`
+function configure(opts) {
+  if (!opts.probes || !opts.probes.length) {
+    return Promise.reject(new Error('At least one urlJobProcessor probe must be specified'));
+  }
+
+  const ret = [Promise.resolve()];
+
+  opts.probes.forEach((probeOpts) => {
+    if (!probeOpts.name) {
+      return Promise.reject(new Error('Probe without name encountered'));
+    }
+
+    // We allow the probe to be specified by name or by path. If specified by name,
+    // we expect it to be a known subdir.
+    const path = probeOpts.path || `./urlJobProcessorProbes/${probeOpts.name}.js`;
+    const probeModule = require(path).default;
+
+    if (!probeModule.runProbe) {
+      return Promise.reject(new Error(`Probe ${probeOpts.name} does not implement runProbe`));
+    }
+
+    let p = Promise.resolve();
+    if (probeModule.configure) {
+      p = probeModule.configure(probeOpts.opts || {});
+    }
+
+    ret.push(p.then(() => {
+      probes.push({
+        name: probeOpts.name,
+        probeFn: probeModule.runProbe,
+      });
+    }));
+  });
+
+  return Promise.all(ret);
```

Comment:
@digitarald I removed your phone number since it didn't seem like you intended to include it. It's part of your email signature so it showed up in your github comment

> If we get concurrency with kue, do we still need it on for the probes?

I don't think we can really know the answer without taking some perf measurements. It probably depends on the types of probes we end up implementing. I don't imagine that it will make too much of a difference and it's a super simple change to make so I'm happy to make them run serially for now and see how things work out.